tools/clang/tools/libclang/dxcrewriteunused.cpp

```diff
@@ -107,7 +112,8 @@ void SetupCompilerForRewrite(CompilerInstance &compiler,
                              _In_ TextDiagnosticPrinter *diagPrinter,
                              _In_opt_ ASTUnit::RemappedFile *rewrite,
                              _In_ hlsl::options::DxcOpts &opts,
-                             _In_opt_ LPCSTR pDefines) {
+                             _In_opt_ LPCSTR pDefines,
+                             _In_opt_ dxcutil::DxcArgsFileSystem *msfPtr) {
```

Comment:
This is fine. I wonder if replacing the static_cast that was there before with a dyn_cast would have had the same effect. Not really a suggestion, just another possibility.

Reply:
Would require RTTI, since this class doesn't have the llvm-style custom type info.  In this case, would rather just pass it down when needed, since it just depends on which API is being used.